docs/DOCUMENTATION.md

```diff
@@ -0,0 +1,857 @@
+# Documentation
+
+- [Oracle Network Structure](#oracle-network-structure)
+- [Smart Contract Details](#smart-contract-details)
+- [Aims/Objectives](#aims/objectives)
+  - [Middleware](#middleware)
+  - [Monitoring](#monitoring)
+- [File structure](#file-structure)
+- [Technical Documentation](#technical-documentation)
+  - [helper.js](#helperjs)
+    - [readJSONFile(filename)](#readJSONFile)
+    - [saveJSONDataToFile(newData, filename)](#saveJSONDataToFile)
+    - [validUrl(url)](#validUrl)
+    - [delay(ms)](#delay)
+  - [middleware.js](#middlewarejs)
+    - [Environment Variables](#envvarsmiddleware)
+    - [readState()](#readState)
+    - [initialiseState()](#initialiseState)
+    - [getJobIndex(jobName)](#getJobIndex)
+    - [sendJobRun(credentials, count, jobId, chainlinkUrl, requestType)](#sendJobRun)
+    - [getOffers(follower)](#getOffers)
+    - [checkSubmissionForRound(oracle, feedOfferId, roundId)](#checkSubmissionForRound)
+    - [queryPrice(feed)](#queryPrice)
+    - [queryRound(feed)](#queryRound)
+    - [submitNewJobIndex(index, requestType)](#submitNewJobIndex)
+    - [makeController()](#makeController)
+    - [getOraclesInvitations()](#getOraclesInvitationsMiddleware)
+    - [pushPrice(price, feed, round, from)](#pushPrice)
+    - [startBridge(port)](#startBridge)
+      - [POST /adapter](#postadapter)
+      - [POST /jobs](#postjobs)
+      - [DELETE /jobs/:id](#deljobs)
+    - [middleware()](#middlewarefunc)
+  - [monitor.js](#monitorjs)
+    - [Environment Variables](#envvarsmonitor)
+    - [Metrics](#metrics)
+    - [readOracleAddresses()](#readOracleAddresses)
+    - [readMonitoringState()](#readMonitoringState)
+    - [getOraclesInvitations()](#getOraclesInvitations)
+    - [updateMetrics(oracleName, oracle, feed, value, id, actualPrice, lastRound)](#updateMetrics)
+    - [updateBalanceMetrics(oracleName, oracle, feed, value)](#updateBalanceMetrics)
+    - [queryPrice(feed)](#queryPriceMonitor)
+    - [getOffersAndBalances(follower, oracle)](#getOffersAndBalances)
+    - [getLatestPrices(oracle, oracleDetails, state)](#getLatestPrices)
+    - [startServer()](#startServer)
+    - [monitor()](#monitorfunc)
+- [Considerations](#considerations)
+  - [Why not use a Chainlink cron job?](#consid1)
+  - [Why does the middleware consist of an exposed API and two intervalled functions?](#consid2)
+  - [Why does the middleware only consider the last few offers when making checks?](#consid3)
+  - [Intervals](#consid4)
+## Oracle Network Structure
+
+<img src="images/struct.png"></img>
+<br>
+
+Each node operator has to run:
+- A Chainlink node hosted locally in Docker that queries Chainlink adapters
```

Comment:
Get a link to what a chainlink adapter is or describe it in a few words

Reply:
https://docs.chain.link/chainlink-nodes/external-adapters/external-adapters/

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/d85360c17cb9f44394ff03ce5df97da7c44ef58a